iocage/cli/fetch.py

```diff
@@ -38,58 +44,114 @@ def validate_count(ctx, param, value):
+              # type=release_choice(),
+              help="The FreeBSD release to fetch.")
+# @click.option("--plugin-file", "-P", is_flag=True,
+#              help="This is a plugin file outside the INDEX, but exists in "
+#                   "that location.\nDeveloper option, most will prefer to "
+#                   "use --plugins.")
+# @click.option("--plugins", help="List all available plugins for creation.",
+#              is_flag=True)
+# @click.argument("props", nargs=-1)
 @click.option("--update/--noupdate", "-U/-NU", default=True,
               help="Decide whether or not to update the fetch to the latest "
                    "patch level.")
-@click.option("--eol/--noeol", "-E/-NE", default=True,
-              help="Enable or disable EOL checking with upstream.")
-@click.option("--name", "-n", help="Supply a plugin name for --plugins to "
-                                   "fetch or use a autocompleted filename"
-                                   " for --plugin-file.\nAlso accepts full"
-                                   " path for --plugin-file.")
-@click.option("--accept/--noaccept", default=False,
-              help="Accept the plugin's LICENSE agreement.")
+# @click.option("--eol/--noeol", "-E/-NE", default=True,
+#               help="Enable or disable EOL checking with upstream.")
+# @click.option("--name", "-n", help="Supply a plugin name for --plugins to "
+#                                    "fetch or use a autocompleted filename"
+#                                    " for --plugin-file.\nAlso accepts full"
+#                                    " path for --plugin-file.")
+# @click.option("--accept/--noaccept", default=False,
+#              help="Accept the plugin's LICENSE agreement.")
+# Compat
+@click.option("--http", "-h", default=False,
+              help="Have --server define a HTTP server instead.", is_flag=True)
+# Compat files
+@click.option("--files", multiple=True,
+              help="Specify the files to fetch from the mirror. "
+              "(Deprecared: renamed to --file)")
+# def cli(url, files, release, update):
 def cli(**kwargs):
-    """CLI command that calls fetch_release()"""
-    ioc.IOCage().fetch(**kwargs)
+
+    release_input = kwargs["release"]
+
+    if isinstance(release_input, int):
+        try:
+            release = host.distribution.releases[int(release_input)]
+        except:
+            logger.error(f"Selection {release_input} out of range")
+            sys.exit(1)
+    else:
+        try:
+            release = Release.Release(
+                name=release_input,
+                host=host,
+                logger=logger
+            )
+        except:
+            self.logger.error(f"Invalid Release '{release_input}'")
+            sys.exit(1)
+
+    url_or_files_selected = False
+
+    # optional --url
+    try:
+        url = kwargs["url"]
+        if url:
+            release.mirror_url = url
+            url_or_files_selected = True
+    except:
+        pass
```

Comment:
I personally do not like code comments. They are a sign of weak interface design and I've used them in some occasions - but only to refactor the code later on, because at the time of writing I couldn't find a more elegant solution.

@araujobsd your feedback that you would require more context reading this is very welcome, but I won't solve it by adding comments - a helper method that checks kwargs for the existence of an attribute would improve the readability in the same way I guess..?